src/hex.ts

```diff
@@ -55,6 +65,59 @@ export function assertIsStrictHexString(value: unknown): asserts value is Hex {
   );
 }
 
+/**
+ * Validate that the passed prefixed hex string is an all-lowercase
+ * hex address, or a valid mixed-case checksum address.
+ *
+ * @param possibleAddress - Input parameter to check against.
+ * @returns Whether or not the input is a valid hex address.
+ */
+export function isValidHexAddress(possibleAddress: Hex) {
+  return (
+    is(possibleAddress, HexAddressStruct) ||
+    isValidChecksumAddress(possibleAddress)
+  );
+}
+
+/**
+ * Encode a passed hex string as an ERC-55 mixed-case checksum address.
+ *
+ * @param address - The hex address to encode.
+ * @returns The address encoded according to ERC-55.
+ * @see https://eips.ethereum.org/EIPS/eip-55
+ */
+export function getChecksumAddress(address: Hex) {
```

Comment:
I wonder if we should do more validation here, such as checking if the provided address is 20 bytes long.

Reply:
I just added an assertion to check the string we receive here